homeassistant/components/flexpool/sensors.py

```diff
@@ -0,0 +1,314 @@
+"""Sensor classes for the Flexpool integration."""
+import flexpoolapi
+
+from homeassistant.const import PERCENTAGE
+from homeassistant.helpers.entity import Entity
+
+from .const import SENSOR_DICT
+from .helper import get_hashrate
+
+
+class FlexpoolBalanceSensor(Entity):
+    """Representation of an Etherscan.io sensor."""
+
+    def __init__(self, name, address):
+        """Initialize the sensor."""
+        info = SENSOR_DICT[name]
+        self._name = info[0]
+        self._icon = info[1]
+        self._address = address
+        self._state = None
+        self._unit_of_measurement = "ETH"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement this sensor expresses itself in."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the latest state of the sensor."""
+        miner = flexpoolapi.miner(self._address)
+        balance = round(miner.balance() / 1000000000000000000, 4)
+
+        self._state = balance
```

Comment:
Each sensor will do its own request. This way, the number of sensors (multiplied by the number of users) can put a real strain on the upstream API.

Please consider implementing a central `DataUpdateCoordinator` that fetches all info for all entities once.

See: <https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities>

Reply:
That's a good point. Workers Sensors may be the most of the flexpool sensors for some users. Will implement that.

Thank you for your review